oml/functional/label_smoothing.py

```diff
@@ -0,0 +1,39 @@
+from typing import Optional
+
+import torch
+import torch.nn.functional as F
+
+
+def label_smoothing(
+    y: torch.Tensor,
+    num_classes: int,
+    epsilon: float = 0.2,
+    categories: Optional[torch.Tensor] = None,
+) -> torch.Tensor:
+    """
+    This function is doing `label smoothing <https://arxiv.org/pdf/1512.00567v3.pdf>`_.
+    You can also use modified version, where the label is smoothed only for the category corresponding to sample's
+    ground truth label. To use this, you should provide the ``categories`` argument: vector, for which i-th entry
+    is a corresponding category for label ``i``.
+
+    Args:
+        y: GT labels
```

Comment:
Let's improve doc string:

Ground truth labels with the size of ``batch_size`` where each element is in the range of ``(0, num_classes - 1)``